packages/tradershub/src/features/cfd/screens/ChangePassword/InvestorPassword/MT5ChangeInvestorPasswordInputsScreen.tsx

```diff
@@ -0,0 +1,116 @@
+                        <TextField
+                            autoComplete='current-password'
+                            className='border-solid rounded-200 border-xs border-system-light-active-background text-75 h-2000'
+                            label='Current investor password'
+                            name='currentPassword'
+                            onChange={handleChange}
+                            value={values.currentPassword}
+                        />
+                        <TextField
+                            autoComplete='new-password'
+                            className='border-solid rounded-200 border-xs border-system-light-active-background text-75 h-2000'
+                            label='New investor password'
+                            name='newPassword'
+                            onChange={handleChange}
+                            value={values.newPassword}
+                        />
+                    </div>
+                    <div className='flex flex-col content-center gap-800'>
+                        <Button
+                            // quill component issue, add type submit and disabled once the proper component is ready
+
+                            isLoading={changeInvestorPasswordStatus === 'loading'}
+                            size={isMobile ? 'lg' : 'md'}
+                        >
+                            Change investor password
+                        </Button>
+                        <Button
+                            className='border-none'
+                            colorStyle='coral'
+                            onClick={sendEmail}
+                            size={isMobile ? 'lg' : 'md'}
+                            variant='secondary'
+                        >
+                            Create or reset investor password
+                        </Button>
+                    </div>
+                </form>
+            )}
+        </Formik>
+    );
+
+    return (
+        <ActionScreen
+            description={
+                <Fragment>
+                    <Text size='sm'>
+                        Use this password to grant viewing access to another user. While they may view your trading
+                        account, they will not be able to trade or take any other actions.
+                    </Text>
+                    <Text size='sm'>
+                        If this is the first time you try to create a password, or you have forgotten your password,
+                        please reset it.
+                    </Text>
+                    {changeInvestorPasswordError && (
+                        <Text align='center' color='error' size='sm'>
+                            {changeInvestorPasswordError?.error?.message}
+                        </Text>
+                    )}
+                </Fragment>
+            }
+            descriptionSize='sm'
+            renderButtons={() => <FormInputs />}
```

Comment:
Just a query about the prop: why is the whole form being passed as a prop and the name of the prop is `renderButtons`. I think it's better to create the form where is it needed than passing it like this

Reply:
The buttons which we require need to be wrapped by `Formik`, That's why it is implemented like this.